### PR TITLE
iterator: Return empty iterator on modification stmt

### PR DESCRIPTION
**Pull request by wprzytula**
...instead of ProtocolError as before. This was requested by #631. Also, added a test for this behaviour.

Fixes: #631

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [x] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

File: scylla/src/transport/iterator.rs [hunk 1]
```diff
@@ -540,6 +540,27 @@ where
...
+                    // We must attempt to send something because the iterator expects it.
+                    let (proof, _) = self
+                        .sender
+                        .send(Ok(ReceivedPage {
+                            rows: Rows {
+                                metadata: Default::default(),
+                                rows_count: 0,
+                                rows: Vec::new(),
+                            },
+                            tracing_id: None,
```
Comment (piodul) [inline scylla/src/transport/iterator.rs:559]: `tracing_id` is available in the `NonErrorQueryResponse`, so let's use it.

**Review by piodul — COMMENTED** [review]
One tiny issue (and a test for it would be nice), otherwise LGTM.

File: scylla/src/transport/iterator.rs (hunk 1 above)
Comment (wprzytula) [reply scylla/src/transport/iterator.rs:559]: How do I test that? (as you mentioned in the general comment)

Comment (piodul) [reply scylla/src/transport/iterator.rs]: Run `query_iter` with tracing enabled (`Query::set_tracing`) and then verify that the list of tracing ids reported by the iterator is not empty.

**Pull request merged**